pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -329,19 +352,26 @@ func (dsw *desiredStateOfWorld) GetVolumesToMount() []VolumeToMount {
 				volumesToMount,
 				VolumeToMount{
 					VolumeToMount: operationexecutor.VolumeToMount{
-						VolumeName:          volumeName,
-						PodName:             podName,
-						Pod:                 podObj.pod,
-						VolumeSpec:          podObj.spec,
-						PluginIsAttachable:  volumeObj.pluginIsAttachable,
-						OuterVolumeSpecName: podObj.outerVolumeSpecName,
-						VolumeGidValue:      volumeObj.volumeGidValue,
-						ReportedInUse:       volumeObj.reportedInUse}})
+						VolumeName:            volumeName,
+						PodName:               podName,
+						Pod:                   podObj.pod,
+						VolumeSpec:            podObj.spec,
+						PluginIsAttachable:    volumeObj.pluginIsAttachable,
+						OuterVolumeSpecName:   podObj.outerVolumeSpecName,
+						VolumeGidValue:        volumeObj.volumeGidValue,
+						ReportedInUse:         volumeObj.reportedInUse,
+						OperationCompleteFunc: volumeObj.operationCompleteFunc}})
 		}
 	}
 	return volumesToMount
 }
 
+func (dsw *desiredStateOfWorld) GetVolumeUnmountMetricCallback(volumeName v1.UniqueVolumeName) func(error) {
+	operationComplete := dsw.volumesToUnmount[volumeName]
+	delete(dsw.volumesToUnmount, volumeName)
+	return operationComplete
```

Comment:
Don't we need to protect this with mutex?

Reply:
Yup. Will add.